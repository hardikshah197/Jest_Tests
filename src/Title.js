import React from 'react';

export default function Title({title, count}) {
    return(
        <div>
            <h1>{title}</h1>
            <h2>{count}</h2>
        </div>
    )
}