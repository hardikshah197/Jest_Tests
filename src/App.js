import logo from './logo.svg';
import './App.css';
import Title from './Title';
import {useState, useEffect } from 'react';

function App() {
  const [count, setCount] = useState(0);
  useEffect(() => {
    if(count > 3)
       return(
        alert('OverLimit')
       )
      else{
        return(
          <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        )
      }
  });
  function countChecker(){
    console.log(count);
    return(setCount(count+1))
  }
  
  return (
    <div className="App">
      <header className="App-header">
        <Title title="My Test Jest Test" count="3"/>
        <img src={logo} className="App-logo" alt="logo" />
        {/* {counterChecker} */}
        <button onClick={countChecker}>Counter</button>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
}

export default App;
