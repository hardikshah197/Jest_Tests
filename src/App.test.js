import { render, screen } from '@testing-library/react';
import App from './App';
import React from 'react';
import { shallow } from 'enzyme';
import Title from './Title';

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});


const title = 'Test Title';
const wrapper = shallow(<Title title={title} />);
const wrapper2 = shallow(<Title count={3} />)

describe('Title', () => {
  it('Snapshot test for title', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('snapshot test for title 2', () => {
    expect(wrapper2).toMatchSnapshot();
  })

  it('Interaction test for title', () => {
    expect(
      wrapper
      .find('h1')
      .text()
    ).toEqual(title);
  });
});