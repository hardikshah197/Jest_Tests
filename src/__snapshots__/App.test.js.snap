// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Snapshot test for title 1`] = `
<div>
  <h1>
    Test Title
  </h1>
  <h2 />
</div>
`;

exports[`Title snapshot test for title 2 1`] = `
<div>
  <h1 />
  <h2>
    3
  </h2>
</div>
`;
